browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,263 @@
+}
+
+void SplitViewLocationBar::OnPaintBorder(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ false);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorBraveSplitViewInactiveWebViewBorder));
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+gfx::Size SplitViewLocationBar::CalculatePreferredSize(
+    const views::SizeBounds& available_size) const {
+  auto size = views::WidgetDelegateView::CalculatePreferredSize(available_size);
+  if (auto* view = view_observation_.GetSource(); view && view->width()) {
+    size.set_width(std::min(view->width(), size.width()));
+  }
+  return size;
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::DidChangeVisibleSecurityState() {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    auto url = contents->GetVisibleURL();
+    auto origin = url::Origin::Create(url);
+    auto formatted_origin = url_formatter::FormatUrl(
+        origin.opaque() ? url : origin.GetURL(),
+        url_formatter::kFormatUrlOmitDefaults |
+            url_formatter::kFormatUrlOmitHTTPS,
+        base::UnescapeRule::SPACES, nullptr, nullptr, nullptr);
+    url_->SetText(formatted_origin);
+    UpdateIcon();
+  } else {
+    url_->SetText({});
+    UpdateIcon();
+  }
+
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::UpdateIcon() {
+  // At the moment, we show only warning icon.
+  safety_icon_->SetVisible(!IsContentsSafe());
+}
+
+bool SplitViewLocationBar::IsContentsSafe() const {
+  if (!web_contents()) {
+    return true;
+  }
+
+  if (SecurityStateTabHelper::FromWebContents(web_contents())
+          ->GetSecurityLevel() == security_state::SecurityLevel::SECURE) {
```

Comment:
what happens if you load https://expired.badssl.com/ (both before and after clicking through the certificate error)?

Reply:
<img width="574" alt="image" src="https://github.com/brave/brave-core/assets/5474642/5f3b258f-1466-48f8-bad0-5619d705ef51">

<img width="574" alt="image" src="https://github.com/brave/brave-core/assets/5474642/6e971067-709f-4076-ab08-b0a510197427">

Reply:
thanks